src/encryption/api_test.go

```diff
@@ -0,0 +1,30 @@
+package encryption
+
+import (
+	"context"
+	"keboola-as-code/src/utils"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewEncryptionApi(t *testing.T) {
+	logger, _ := utils.NewDebugLogger()
+	a := NewEncryptionApi("connection.keboola.com", context.Background(), logger, true)
+	assert.NotNil(t, a)
+	assert.Equal(t, "encryption.keboola.com", a.apiHost)
+	mapToEncrypt := map[string]string{"#keyToEncrypt": "value"}
+	encryptedMap, err := a.EncryptMapValues("keboola.ex-generic-v2", "1234", mapToEncrypt)
+	assert.Nil(t, err)
+	assert.NotNil(t, encryptedMap)
+	assert.True(t, isEncrypted(encryptedMap["#keyToEncrypt"]))
+}
+
+func TestErrorEncryptionApi(t *testing.T) {
+	logger, _ := utils.NewDebugLogger()
+	mapToEncrypt := map[string]string{"#keyToEncrypt": "value"}
+	a := NewEncryptionApi("connection.keboola.com", context.Background(), logger, false)
+	_, err := a.EncryptMapValues("", "", mapToEncrypt)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "The componentId parameter is required")
+}
```

Comment:
Mozno tu tie testy by mali byt ako funkcionalne niekde inde kedze sa vola priamo encryption api. Mocking api som neriesil.

Reply:
myslim, ze je to v pohode testovat priamo oproti API v tomto pripade.

Reply:
k vytvaraniu novej instancie Storage API som uz pisal ... v teste by tu potom malo byt `NewEncryptionApi("https://encryption.keboola.com", ...)`

Reply:
A ako som pisal, teraz je to lepsie testovatelne. Nemusis vytvarat `StorageApi`, testuje sa len jedna vec.

Samotne ziskanie `encryption host` bude mat zas vlastny unit test, v `remote` balicku.

Reply:
ja som to pochopil ze je to tu ok testovat oproti api obecne. Kazdopadne upravil som to, teda separoval nacitanie encryption api url do StorageAPI

Reply:
jj, mne nevadilo, ze sa volalo StorageAPI v teste (cele StorageApi sa testuj proti realnemu api, az na nejake vynimky, ked sa simuluju chyby) ... islo mi cisto o tie zavislosti.